### PR TITLE
4 como cliente quiero crear mi propia cartera

### LINKED ISSUE #4: Como cliente quiero crear mi propia cartera
**Descripción**
Recibo una petición POST y creo una nueva cartera.

**Ejemplo**

Servicio no disponible:
Request: POST /wallet/open
```
{ 
}
```

Response: 503 Service Unavailable
```
{
  "error": "service is unavailable"
}
```

Happy Path:
Request: POST /wallet/open
```
{ 
}
```

Response: 200 OK
```
{
  "wallet_id": "string"
}
```

### PR DESCRIPTION
**Pull request by raulsc-cl**
Pull request for issue #4

### CODE REVIEW

File: app/Application/CreateWallet/CreateWalletService.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        $this->cryptoDataStorage = $cryptoDataStorage;
+    }
+
+    /**
+     * @return string
+     * @throws Exception
+     */
+    public function createWallet(): string
+    {
+        $user_wallet = $this->cryptoDataStorage->createWallet();
```
Comment (xAriextz) [inline app/Application/CreateWallet/CreateWalletService.php:30]: Consider changing one of the methods name

File: app/Infrastructure/Controllers/CreateWalletController.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+     */
+    public function __construct(CreateWalletService $createWalletService)
+    {
+        $this->createWalletService = $createWalletService;
+    }
+
+    public function __invoke(): JsonResponse
+    {
+        try {
+            $wallet_id = $this->createWalletService->createWallet();
```
Comment (xAriextz) [inline app/Infrastructure/Controllers/CreateWalletController.php]: Maybe change variable name to $userWalletId

File: tests/app/Application/CreateWallet/CreateWalletServiceTest.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+        $this->cryptoDataStorage = Mockery::mock(CryptoDataStorage::class);
+
+        $this->createWalletService = new CreateWalletService($this->cryptoDataStorage);
+    }
+
+    /**
+     * @test
+     */
+
```
Comment (xAriextz) [inline tests/app/Application/CreateWallet/CreateWalletServiceTest.php]: Delete line

**Review by xAriextz — CHANGES_REQUESTED** [review]
Consider making changes based on my feedback

File: app/Application/CreateWallet/CreateWalletService.php (hunk 1 above)
Comment (jesuslezaun) [reply app/Application/CreateWallet/CreateWalletService.php:30]: method in CreateWalletService could be called just execute()

File: tests/app/Application/CreateWallet/CreateWalletServiceTest.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            ->andThrow(new Exception('Service unavailable'));
+
+        $this->expectException(Exception::class);
+
+        $this->createWalletService->createWallet();
+    }
+
+    /**
+     * @test
+     * @throws Exception
```
Comment (jesuslezaun) [inline tests/app/Application/CreateWallet/CreateWalletServiceTest.php]: Delete throws Exception

**Review by jesuslezaun — CHANGES_REQUESTED** [review]
Please consider the changes request made

**Review by jesuslezaun — APPROVED** [review]
Ok to merge

**Review by xAriextz — APPROVED** [review]
Ok to merge

**Pull request merged**